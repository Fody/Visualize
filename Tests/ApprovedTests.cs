using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using NUnit.Framework;

[UseReporter(typeof(DiffReporter))]
#if(DEBUG)
[UseApprovalSubdirectory("results-debug")]
#else
[UseApprovalSubdirectory("results-release")]
#endif
[TestFixture]
public class ApprovedTests
{
    [Test]
    public void ClassWithProperties()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithProperties"));
    }

    [Test]
    public void ClassWithConst()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithConst"));
    }

    [Test]
    public void ClassWithIndexor()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIndexor"));
    }

    [Test]
    public void ClassWithDataAnnotations()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithDataAnnotations"));
    }

    [Test]
    public void ClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerable"));
    }

    [Test]
    public void ClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithICollection"));
    }

    [Test]
    public void GenericClassWithIEnumerable()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithIEnumerable`1"));
    }

    [Test]
    public void GenericClassWithICollection()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.GenericClassWithICollection`1"));
    }

    [Test]
    public void ClassWithIEnumerableNotAttributed()
    {
        Approvals.Verify(Decompiler.Decompile(AssemblyWeaver.AfterAssemblyPath, "AssemblyToProcess.ClassWithIEnumerableNotAttributed"));
    }

    //[Test]
    //public void Sample()
    //{
    //    var sampleClassType = AssemblyWeaver.Assembly.GetType("AssemblyToProcess.ClassWithIEnumerable");
    //    var sample = Activator.CreateInstance(sampleClassType);
    //}
}