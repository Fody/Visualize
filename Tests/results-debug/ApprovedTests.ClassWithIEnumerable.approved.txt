.class public auto ansi beforefieldinit AssemblyToProcess.ClassWithIEnumerable
       extends [mscorlib]System.Object
       implements class [mscorlib]System.Collections.Generic.IEnumerable`1<int32>,
                  [mscorlib]System.Collections.IEnumerable
{
  .custom instance void [mscorlib]System.Diagnostics.DebuggerTypeProxyAttribute::.ctor(class [mscorlib]System.Type) = ( 01 00 42 41 73 73 65 6D 62 6C 79 54 6F 50 72 6F   // ..BAssemblyToPro
                                                                                                                        63 65 73 73 2E 43 6C 61 73 73 57 69 74 68 49 45   // cess.ClassWithIE
                                                                                                                        6E 75 6D 65 72 61 62 6C 65 2B 3C 43 6C 61 73 73   // numerable+<Class
                                                                                                                        57 69 74 68 49 45 6E 75 6D 65 72 61 62 6C 65 3E   // WithIEnumerable>
                                                                                                                        50 72 6F 78 79 00 00 )                            // Proxy..
  .class auto ansi sealed nested private beforefieldinit '<ClassWithIEnumerable>Proxy'
         extends [mscorlib]System.Object
  {
    .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = ( 01 00 00 00 ) 
    .field private initonly class AssemblyToProcess.ClassWithIEnumerable original
    .method public hidebysig specialname rtspecialname 
            instance void  .ctor(class AssemblyToProcess.ClassWithIEnumerable original) cil managed
    {
      // Code size       14 (0xe)
      .maxstack  8
      IL_0000:  ldarg.0
      IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
      IL_0006:  ldarg.0
      IL_0007:  ldarg.1
      IL_0008:  stfld      class AssemblyToProcess.ClassWithIEnumerable AssemblyToProcess.ClassWithIEnumerable/'<ClassWithIEnumerable>Proxy'::original
      IL_000d:  ret
    } // end of method '<ClassWithIEnumerable>Proxy'::.ctor
    .method public hidebysig specialname 
            instance int32[]  get_Items() cil managed
    {
      // Code size       17 (0x11)
      .maxstack  8
      IL_0000:  ldarg.0
      IL_0001:  ldfld      class AssemblyToProcess.ClassWithIEnumerable AssemblyToProcess.ClassWithIEnumerable/'<ClassWithIEnumerable>Proxy'::original
      IL_0006:  newobj     instance void class [mscorlib]System.Collections.Generic.List`1<int32>::.ctor(class [mscorlib]System.Collections.Generic.IEnumerable`1<!0>)
      IL_000b:  call       instance !0[] class [mscorlib]System.Collections.Generic.List`1<int32>::ToArray()
      IL_0010:  ret
    } // end of method '<ClassWithIEnumerable>Proxy'::get_Items
    .property instance int32[] Items()
    {
      .custom instance void [mscorlib]System.Diagnostics.DebuggerBrowsableAttribute::.ctor(valuetype [mscorlib]System.Diagnostics.DebuggerBrowsableState) = ( 01 00 03 00 00 00 00 00 ) 
      .get instance int32[] AssemblyToProcess.ClassWithIEnumerable/'<ClassWithIEnumerable>Proxy'::get_Items()
    } // end of property '<ClassWithIEnumerable>Proxy'::Items
  } // end of class '<ClassWithIEnumerable>Proxy'
  .method public hidebysig newslot virtual final 
          instance class [mscorlib]System.Collections.Generic.IEnumerator`1<int32> 
          GetEnumerator() cil managed
  {
    // Code size       19 (0x13)
    .maxstack  2
    .locals init ([0] class [mscorlib]System.Collections.Generic.IEnumerator`1<int32> )
    .language '{3F5162F8-07C6-11D3-9053-00C04FA302A1}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
    .line 11,11 : 9,10 'AssemblyToProcess\\ClassWithIEnumerable.cs'
    IL_0000:  nop
    .line 12,12 : 13,60 ''
    IL_0001:  ldc.i4.0
    IL_0002:  ldc.i4.s   10
    IL_0004:  call       class [mscorlib]System.Collections.Generic.IEnumerable`1<int32> [System.Core]System.Linq.Enumerable::Range(int32,
                                                                                                                                    int32)
    IL_0009:  callvirt   instance class [mscorlib]System.Collections.Generic.IEnumerator`1<!0> class [mscorlib]System.Collections.Generic.IEnumerable`1<int32>::GetEnumerator()
    IL_000e:  stloc.0
    IL_000f:  br.s       IL_0011
    .line 13,13 : 9,10 ''
    IL_0011:  ldloc.0
    IL_0012:  ret
  } // end of method ClassWithIEnumerable::GetEnumerator
  .method private hidebysig newslot virtual final 
          instance class [mscorlib]System.Collections.IEnumerator 
          System.Collections.IEnumerable.GetEnumerator() cil managed
  {
    .override [mscorlib]System.Collections.IEnumerable::GetEnumerator
    // Code size       12 (0xc)
    .maxstack  1
    .locals init ([0] class [mscorlib]System.Collections.IEnumerator )
    .line 16,16 : 9,10 ''
    IL_0000:  nop
    .line 17,17 : 13,36 ''
    IL_0001:  ldarg.0
    IL_0002:  call       instance class [mscorlib]System.Collections.Generic.IEnumerator`1<int32> AssemblyToProcess.ClassWithIEnumerable::GetEnumerator()
    IL_0007:  stloc.0
    IL_0008:  br.s       IL_000a
    .line 18,18 : 9,10 ''
    IL_000a:  ldloc.0
    IL_000b:  ret
  } // end of method ClassWithIEnumerable::System.Collections.IEnumerable.GetEnumerator
  .method public hidebysig specialname rtspecialname 
          instance void  .ctor() cil managed
  {
    // Code size       8 (0x8)
    .maxstack  8
    IL_0000:  ldarg.0
    IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
    IL_0006:  nop
    IL_0007:  ret
  } // end of method ClassWithIEnumerable::.ctor
} // end of class AssemblyToProcess.ClassWithIEnumerable